pkg/controller/network/resourcerecordset/controller.go

```diff
@@ -0,0 +1,158 @@
+package resourcerecordset
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/route53"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	"github.com/crossplane/provider-aws/apis/network/v1alpha3"
+	"github.com/crossplane/provider-aws/pkg/clients/resourcerecordset"
+	"github.com/crossplane/provider-aws/pkg/controller/utils"
+)
+
+const (
+	errUnexpectedObject = "The managed resource is not an ResourceRecordSet resource"
+	errList             = "failed to list the ResourceRecordSet resource"
+	errCreate           = "failed to create the ResourceRecordSet resource"
+	errUpdate           = "failed to update the ResourceRecordSet resource"
+	errDelete           = "failed to delete the ResourceRecordSet resource"
+)
+
+// SetupResourceRecordSet adds a controller that reconciles ResourceRecordSets.
+func SetupResourceRecordSet(mgr ctrl.Manager, l logging.Logger) error {
+	name := managed.ControllerName(v1alpha3.ResourceRecordSetGroupKind)
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha3.ResourceRecordSet{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha3.ResourceRecordSetGroupVersionKind),
+			managed.WithExternalConnecter(&connector{client: mgr.GetClient(), newClientFn: resourcerecordset.NewClient, awsConfigFn: utils.RetrieveAwsConfigFromProvider}),
+			managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(mgr.GetClient())),
+			managed.WithConnectionPublishers(),
+			managed.WithInitializers(),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connector struct {
+	client      client.Client
+	newClientFn func(*aws.Config) resourcerecordset.Client
+	awsConfigFn func(context.Context, client.Reader, *corev1.ObjectReference) (*aws.Config, error)
+}
+
+func (conn *connector) Connect(ctx context.Context, mgd resource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mgd.(*v1alpha3.ResourceRecordSet)
+	if !ok {
+		return nil, errors.New(errUnexpectedObject)
+	}
+
+	awsconfig, err := conn.awsConfigFn(ctx, conn.client, cr.Spec.ProviderReference)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
https://github.com/crossplane/provider-aws/blob/2117914/pkg/controller/cache/cachesubnetgroup/controller.go#L90

Do we handle the case where we want to authenticate with a service account, like the above example?

Reply:
Yes. Since we are using the [utils.RetrieveAwsConfigFromProvider](https://github.com/crossplane/provider-aws/pkg/controller/utils/utils.go#L38) function to get our `*aws.Config` the same has been taken care of in the mentioned function itself.